webapp/components/user_settings/email_notification_setting.jsx

```diff
@@ -37,9 +37,16 @@ export default class EmailNotificationSetting extends React.Component {
                 emailInterval: PreferenceStore.getInt(Preferences.CATEGORY_NOTIFICATIONS, Preferences.EMAIL_INTERVAL, Preferences.INTERVAL_FIFTEEN_MINUTES)
             };
         } else {
-            // otherwise, the default interval is immediately
+            /**
+             * otherwise, the default interval is immediately
+             *
+             * `emailInterval` in the backend is irrespective of whether email Bacthing is enabled or not.
+             * Hence this check is kept in the component to choose the default interval as `immediately` when
+             * `EnableEmailBatching` is false.
+             */
+
             this.state = {
-                emailInterval: PreferenceStore.getInt(Preferences.CATEGORY_NOTIFICATIONS, Preferences.EMAIL_INTERVAL, Preferences.INTERVAL_IMMEDIATE)
+                emailInterval: Preferences.INTERVAL_IMMEDIATE
```

Comment:
Hey @MusikPolice 

Ya i had seen that part. But since when email batching is disabled. there are only 2 choices. `Immediate` or `never`.

By default when email batching is disabled, `emailInterval` is set to Preferences.INTERVAL_IMMEDIATE.

And in the component when checked for the `Immediate` or `never` radio is determined, i.e
`checked={this.props.enableEmail && this.state.emailInterval === Preferences.INTERVAL_IMMEDIATE}` for immediate and `checked={!this.props.enableEmail}` for never. 

It is based on `this.props.enableEmail`. Hence the if the user selects `never`. this.props.enableEmail will be false, hence never will be selected, even if the email dispatching is disabled.

Let me know if I am missing something. Or there should be another approach for this.